### PR TITLE
Create Video UiElement

### PR DESCRIPTION
**Pull request by RedHotMan**
Issue [8480](https://support.monsieurbiz.com/issues/8480)

### What does it do:
- Create form for Video UiElement
- Create Video UiElement
- Handle translations
- Add Video UiElement to the plugin config
- Update model to display a video
- Create Video UiElement template

### Screenshot:
Capture d’écran 2019-12-13 à 16 13 15
Capture d’écran 2019-12-13 à 16 13 47

### CODE REVIEW

File: src/Resources/translations/messages.en.yml [hunk 1]
```diff
@@ -22,6 +22,13 @@ monsieurbiz_cmsplugin:
       field:
         author: 'Author'
         content: 'Content'
+    video:
+      title: 'Lame video'
```
Comment (jacquesbh) [inline src/Resources/translations/messages.en.yml]:
```suggestion
      title: 'Lame vidéo'
```

File: src/Resources/translations/messages.en.yml [hunk 2]
```diff
@@ -22,6 +22,13 @@ monsieurbiz_cmsplugin:
...
+      short_description: 'Description courte lame video'
```
Comment (jacquesbh) [inline src/Resources/translations/messages.en.yml]: Could you write a real short description please?

File: src/Resources/translations/messages.en.yml [hunk 3]
```diff
@@ -22,6 +22,13 @@ monsieurbiz_cmsplugin:
...
+      description: 'Une lame contenant une video'
+      field:
+        video: 'Video'
```
Comment (jacquesbh) [inline src/Resources/translations/messages.en.yml]:
```suggestion
        video: 'Vidéo'
```

File: src/Resources/translations/messages.en.yml [hunk 4]
```diff
@@ -22,6 +22,13 @@ monsieurbiz_cmsplugin:
       field:
         author: 'Author'
         content: 'Content'
+    video:
+      title: 'Lame video'
+      short_description: 'Description courte lame video'
+      description: 'Une lame contenant une video'
```
Comment (jacquesbh) [inline src/Resources/translations/messages.en.yml]: This description is shorter than the short description :).

File: src/Form/Type/UiElement/VideoType.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+use Symfony\Component\Validator\Constraints as Assert;
+
+class VideoType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('video', FileType::class, [
+                'label' => 'monsieurbiz_cmsplugin.ui_element.video.field.video',
+                'data_class' => null, // @TODO check to have the File original data class, and remove this line
```
Comment (jacquesbh) [inline src/Form/Type/UiElement/VideoType.php]: Why do you still have this TODO?

File: src/Form/Type/UiElement/VideoType.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                        'mimeTypes' => [
+                            'video/mp4',
+                            'video/mp3'
+                        ]
+                    ])
+                ],
+            ])
+            ->add('image', FileType::class, [
+                'label' => 'monsieurbiz_cmsplugin.ui_element.video.field.image',
+                'data_class' => null, // @TODO check to have the File original data class, and remove this line
```
Comment (jacquesbh) [inline src/Form/Type/UiElement/VideoType.php]: Why do you still have this TODO?

Comment (RedHotMan) [reply src/Form/Type/UiElement/VideoType.php]: I have an error if I remove the `data_class => null` option. 
```
The form's view data is expected to be an instance of class Symfony\Component\HttpFoundation\File\File, but is a(n) string. You can avoid this error by setting the "data_class" option to null or by adding a view transformer that transforms a(n) string to an instance of Symfony\Component\HttpFoundation\File\File
```

This is due to the fact that the field expect a file instance. When uploading there is no problem, but when updating, I only get the file path from the database so it's a string and not a file instance. 
I looked for a solution but I didn't found any. The only thing we can do is to define `data_class` option to `null`. See this [Stackoverflow comment ](https://stackoverflow.com/a/54633281/11398936)

**Pull request merged**